src/XTMF2/Controllers/ProjectController.cs

```diff
@@ -74,6 +74,58 @@ public bool CreateNewProject(User owner, string name, out ProjectSession session
             }
         }
 
+        /// <summary>
+        /// Import a project file.
+        /// </summary>
+        /// <param name="owner">The user that will own this project.</param>
+        /// <param name="name">The name to give this project. This must be unique for the given user.</param>
+        /// <param name="filePath">The path to the project file.</param>
+        /// <param name="session">An editing session for the imported project.</param>
+        /// <param name="error">An error message if the operation fails.</param>
+        /// <returns>True if the operation completes successfully, false otherwise with an error message.</returns>
+        public bool ImportProjectFile(User owner, string name, string filePath, out ProjectSession session, ref string error)
+        {
+            session = null;
+            if (owner is null)
+            {
+                throw new ArgumentNullException(nameof(owner));
+            }
+
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                error = "A project must have a non-blank unique name.";
+                return false;
+            }
+            if (string.IsNullOrWhiteSpace(filePath))
+            {
+                throw new ArgumentException("message", nameof(filePath));
```

Comment:
Should the message be replaced with the information about it being null / whitespace?

Reply:
Good catch.